In [41]:
import pandas as pd

In [2]:
data = pd.read_csv('Assignment Log.csv')

In [3]:
data.head()

Event occurred at                           Analyst  \
0  06/22/2017 19:59:06  9fcbc63ff4c8bea5cea4efad782c87cf   
1  06/22/2017 19:59:02  9fcbc63ff4c8bea5cea4efad782c87cf   
2  06/22/2017 19:51:30  85c7b78e76b5232cd38014ea4cdc8f56   
3  06/22/2017 19:51:01  0e9802516f8a79dd0d45211dd4ee74af   
4  06/22/2017 19:50:58  85c7b78e76b5232cd38014ea4cdc8f56   

   Quality score (sourcing)  Quality score (writing)        Action  \
0                      5.00                     5.00  Accepted Job   
1                      5.00                     5.00  Assigned Job   
2                      4.35                     4.35  Declined Job   
3                      4.50                     4.50  Accepted Job   
4                      4.35                     4.35  Assigned Job   

                    Request   Request created at       Job  Wait time (min)  \
0  594bec5c95e2ce005840c23a  06/22/2017 12:12:12    review                1   
1  594bec5c95e2ce005840c23a  06/22/2017 12:12:12    review                1   
2  594bec83fd2cf400280aa965  06/22/2017 12:12:51   writing                9   
3  594c1f5cd7e68f0028c9062c  06/22/2017 15:49:48  sourcing                1   
4  594bec83fd2cf400280aa965  06/22/2017 12:12:51   writing                8   

         Waiting for  Analysts available  Analysts occupied  \
0             review                   0                 13   
1             review                   1                 13   
2  sourcing, writing                   1                 12   
3  sourcing, writing                   1                 11   
4  sourcing, writing                   2                 11   

   Total jobs available  Review jobs available  Vetting jobs available  \
0                    14                      4                       6   
1                    15                      5                       6   
2                    12                      5                       5   
3                    12                      5                       5   
4                    14                      5                       5   

   Planning jobs available  Editing jobs available  Sourcing jobs available  \
0                        2                       0                        1   
1                        2                       0                        1   
2                        1                       0                        0   
3                        1                       0                        0   
4                        2                       0                        1   

   Writing jobs available  
0                       1  
1                       1  
2                       1  
3                       1  
4                       1

In [4]:
data.columns

Index(['Event occurred at', 'Analyst', 'Quality score (sourcing)',
       'Quality score (writing)', 'Action', 'Request', 'Request created at',
       'Job', 'Wait time (min)', 'Waiting for', 'Analysts available',
       'Analysts occupied', 'Total jobs available', 'Review jobs available',
       'Vetting jobs available', 'Planning jobs available',
       'Editing jobs available', 'Sourcing jobs available',
       'Writing jobs available'],
      dtype='object')

In [5]:
data.describe()

Quality score (sourcing)  Quality score (writing)  Wait time (min)  \
count                791.000000               791.000000       791.000000   
mean                   3.935297                 3.935297         5.099874   
std                    1.869956                 1.869956        12.153611   
min                    0.000000                 0.000000         1.000000   
25%                    4.400000                 4.400000         1.000000   
50%                    4.830000                 4.830000         1.000000   
75%                    5.000000                 5.000000         2.000000   
max                    5.000000                 5.000000       107.000000   

       Analysts available  Analysts occupied  Total jobs available  \
count          791.000000         791.000000            791.000000   
mean             2.673831          15.214918              8.054362   
std              1.870439           4.868065              3.644080   
min              0.000000           7.000000              0.000000   
25%              1.000000          11.000000              5.500000   
50%              2.000000          14.000000              8.000000   
75%              4.000000          20.000000             11.000000   
max              8.000000          25.000000             16.000000   

       Review jobs available  Vetting jobs available  Planning jobs available  \
count             791.000000              791.000000               791.000000   
mean                2.836915                1.628319                 1.737042   
std                 2.802793                1.627581                 2.841667   
min                 0.000000                0.000000                 0.000000   
25%                 0.000000                1.000000                 0.000000   
50%                 2.000000                1.000000                 0.000000   
75%                 5.000000                2.000000                 2.000000   
max                11.000000                7.000000                12.000000   

       Editing jobs available  Sourcing jobs available  Writing jobs available  
count              791.000000               791.000000              791.000000  
mean                 0.087231                 0.385588                1.379267  
std                  0.295496                 0.619714                1.547648  
min                  0.000000                 0.000000                0.000000  
25%                  0.000000                 0.000000                0.000000  
50%                  0.000000                 0.000000                1.000000  
75%                  0.000000                 1.000000                2.000000  
max                  2.000000                 4.000000                6.000000

---------------------------------------------------
The average score of analysts is 3.9/5 for both sourcing and writing.
However, the median is a better indicator of overall analyst quality since the quartiles show that the distribution of scores is heavily left skewed. That means that most of the Analysts are highly rated.

Each analyst seems to have the same score for both sourcing and writing, can condense this to one rating or define clearer definitions of sourcing and writing.

Maximum wait time was 107 minutes, the minimum was 1 minute and the average is 5 minutes

There is around 2~3 analysts available at any given time to accept a job.

The number of free analysts peaked at 8.

Average of 15 Analysts working at once.

The average number of available jobs is higher than the average number of available analysts, so more analysts might need to be hired to keep up with the in-flow of new jobs.

---------------------------------------------------

In [55]:
data['Analyst'].nunique()

71

In [56]:
data['Request'].nunique()

74

----------------------------------------------------
There are 71 Analysts and 74 Different Jobs available at this given time frame

--------------------------------------

In [59]:
data[data['Action']=='Accepted Job']['Job'].value_counts()

sourcing         74
review           65
writing          54
vetting          35
planning         33
source review    27
editing          15
Name: Job, dtype: int64

**Looking only at jobs that were accepted,**
* 74 were sourcing
* 65 were review
* 54 were writing
* 35 were vetting
* 33 were planning
* 27 were source review
* 15 were editing

Can hire more or less analysts of certain roles to match this distribution

## Changed strings to DateTime Objects

In [6]:
data.dtypes

Event occurred at            object
Analyst                      object
Quality score (sourcing)    float64
Quality score (writing)     float64
Action                       object
Request                      object
Request created at           object
Job                          object
Wait time (min)               int64
Waiting for                  object
Analysts available            int64
Analysts occupied             int64
Total jobs available          int64
Review jobs available         int64
Vetting jobs available        int64
Planning jobs available       int64
Editing jobs available        int64
Sourcing jobs available       int64
Writing jobs available        int64
dtype: object

In [10]:
data['Event occurred at'] = pd.to_datetime(data['Event occurred at'])

In [12]:
data['Request created at'] = pd.to_datetime(data['Request created at'])

In [45]:
data.sort_values('Event occurred at',inplace=True)

# Checking longest time between Job Creation and Job Accepted

In [46]:
data['Action'].unique()

array(['Assigned Job', 'Accepted Job', 'Declined Job'], dtype=object)

In [47]:
time_diff = data[data['Action'] == 'Accepted Job']

In [48]:
(time_diff['Event occurred at']-time_diff['Request created at']).sort_values()

263   0 days 00:00:35
96    0 days 00:00:39
354   0 days 00:01:27
358   0 days 00:02:03
225   0 days 00:02:28
111   0 days 00:07:14
146   0 days 00:07:48
299   0 days 00:09:27
102   0 days 00:09:56
691   0 days 00:12:38
104   0 days 00:15:43
679   0 days 00:21:49
295   0 days 00:22:45
289   0 days 00:28:29
126   0 days 00:28:29
787   0 days 00:28:32
124   0 days 00:28:59
287   0 days 00:29:43
668   0 days 00:29:48
667   0 days 00:29:49
285   0 days 00:31:33
665   0 days 00:33:02
584   0 days 00:33:23
253   0 days 00:46:41
659   0 days 00:48:54
649   0 days 00:52:26
777   0 days 00:52:38
106   0 days 00:53:22
644   0 days 00:54:10
142   0 days 00:54:11
            ...      
789   0 days 23:47:09
779   0 days 23:56:23
785   1 days 01:25:11
748   1 days 02:15:19
729   1 days 02:24:36
705   1 days 02:37:37
629   1 days 04:05:33
557   1 days 06:02:32
510   1 days 07:08:59
728   1 days 09:22:29
721   1 days 09:26:32
689   1 days 09:36:23
685   1 days 10:04:36
418   1 days 10:40:42
403   1 da

**Longest time was 2 Days and 11 Hours while the shortest was less than 1 minute**

In [49]:
data[data.Request=='59485c262e71030033104e3c']

Event occurred at                           Analyst  \
720 2017-06-21 21:26:17  b599bfb42906772db81ac90137fc1916   
716 2017-06-21 21:26:19  b599bfb42906772db81ac90137fc1916   
532 2017-06-22 01:39:30  a09c8906073b4c0b75e3100b857b982a   
531 2017-06-22 01:39:39  a09c8906073b4c0b75e3100b857b982a   
451 2017-06-22 04:50:59  7e22ad15724c44543d1d4bcafd10c812   
450 2017-06-22 04:51:04  7e22ad15724c44543d1d4bcafd10c812   
438 2017-06-22 05:04:00  e817dd4305458b293cbeb3015da99565   
437 2017-06-22 05:04:04  e817dd4305458b293cbeb3015da99565   
389 2017-06-22 06:19:33  642782c690c8d963c487300a4751e220   
388 2017-06-22 06:19:35  642782c690c8d963c487300a4751e220   

     Quality score (sourcing)  Quality score (writing)        Action  \
720                      0.00                     0.00  Assigned Job   
716                      0.00                     0.00  Accepted Job   
532                      5.00                     5.00  Assigned Job   
531                      5.00                     5.00  Accepted Job   
451                      5.00                     5.00  Assigned Job   
450                      5.00                     5.00  Accepted Job   
438                      4.55                     4.55  Assigned Job   
437                      4.55                     4.55  Accepted Job   
389                      4.97                     4.97  Assigned Job   
388                      4.97                     4.97  Accepted Job   

                      Request  Request created at            Job  \
720  59485c262e71030033104e3c 2017-06-19 19:20:06       sourcing   
716  59485c262e71030033104e3c 2017-06-19 19:20:06       sourcing   
532  59485c262e71030033104e3c 2017-06-19 19:20:06  source review   
531  59485c262e71030033104e3c 2017-06-19 19:20:06  source review   
451  59485c262e71030033104e3c 2017-06-19 19:20:06        writing   
450  59485c262e71030033104e3c 2017-06-19 19:20:06        writing   
438  59485c262e71030033104e3c 2017-06-19 19:20:06        writing   
437  59485c262e71030033104e3c 2017-06-19 19:20:06        writing   
389  59485c262e71030033104e3c 2017-06-19 19:20:06         review   
388  59485c262e71030033104e3c 2017-06-19 19:20:06         review   

     Wait time (min)                                   Waiting for  \
720                1                                      sourcing   
716                1                                      sourcing   
532                1  review, planning, editing, sourcing, writing   
531                1  review, planning, editing, sourcing, writing   
451                1                             sourcing, writing   
450                1                             sourcing, writing   
438                1                             sourcing, writing   
437                1                             sourcing, writing   
389                1   review, vetting, editing, sourcing, writing   
388                1   review, vetting, editing, sourcing, writing   

     Analysts available  Analysts occupied  Total jobs available  \
720                   4                 21                     8   
716                   2                 21                     6   
532                   2                 16                    13   
531                   2                 16                    12   
451                   1                 14                     7   
450                   0                 14                     6   
438                   2                 14                     6   
437                   1                 14                     5   
389                   1                  9                     7   
388                   0                  9                     6   

     Review jobs available  Vetting jobs available  Planning jobs available  \
720                      5                       1                        0   
716                      5                       1                        0   
532                      6         

# Finding the longest time between Assigned job and Accepted job

In [69]:
temp = data[data['Action'] == 'Accepted Job'][['Analyst','Request','Event occurred at']]

In [96]:
for r in temp.iterrows():
    temp.at[r[0],'Assigned During'] = data[(data['Action']=='Assigned Job')&(data['Analyst']==r[1]['Analyst'])&(data['Request']==r[1]['Request'])]['Event occurred at'].max()

In [99]:
temp.rename(columns={'Event occurred at':'Accepted During'},inplace=True)

In [103]:
temp['Wait Time'] = temp['Accepted During']-temp['Assigned During']

In [126]:
temp.sort_values('Wait Time')

Analyst                   Request  \
741  9fcbc63ff4c8bea5cea4efad782c87cf  594ae6fb4f06e20035d92345   
360  642782c690c8d963c487300a4751e220  594b945bf8b866004036d9c3   
681  a09c8906073b4c0b75e3100b857b982a  594a8f5664251c002831f4c5   
593  9fcbc63ff4c8bea5cea4efad782c87cf  594b20ed3f82d10028f03c6c   
235  732e4a29b2e318bce3c1e923276ddf51  594bec5b95e2ce005840c232   
91   a09c8906073b4c0b75e3100b857b982a  594bec5b95e2ce005840c232   
248  732e4a29b2e318bce3c1e923276ddf51  594bec5a95e2ce005840c22a   
478  6c6bd5e891c4015e534773dfb19f29b0  594b52f717d10a00289676dd   
476  6c6bd5e891c4015e534773dfb19f29b0  594b52f717d10a00289676dd   
472  6c6bd5e891c4015e534773dfb19f29b0  594b52f717d10a00289676dd   
146  62060850630c7afe54fd59151413d237  594c15a573dd9c002873cda1   
785  39012c98c8fb80752d2bbcc3dc285230  5949a81c4d1319005556396c   
124  62060850630c7afe54fd59151413d237  594c15a573dd9c002873cda1   
433  6c6bd5e891c4015e534773dfb19f29b0  594b6c6a96165a002896b124   
431  6c6bd5e891c4015e534773dfb19f29b0  594b6c6a96165a002896b124   
426  642782c690c8d963c487300a4751e220  594af29d3f82d10028f03bb2   
446  7e22ad15724c44543d1d4bcafd10c812  594b0303f3b9f600287eac2b   
19   710dbda77cade1eced9c735c08e6e38d  594c41ab1e24bf00350a1f11   
564  008fa95f5c94985e2d44047aeac31655  594a5e5566cb69006fdcf09d   
403  642782c690c8d963c487300a4751e220  5949a81c4d1319005556396c   
562  008fa95f5c94985e2d44047aeac31655  594a5e5566cb69006fdcf09d   
561  008fa95f5c94985e2d44047aeac31655  594a5e5566cb69006fdcf09d   
63   732e4a29b2e318bce3c1e923276ddf51  594b945bf8b866004036d9c3   
757  0e9802516f8a79dd0d45211dd4ee74af  594ab554c864e200288e76ef   
697  a09c8906073b4c0b75e3100b857b982a  59480d57e759070028da6467   
695  a09c8906073b4c0b75e3100b857b982a  594ae6fa4f06e20035d9233d   
504  c033808b89b50e6f550caf266470f090  594ab554c864e200288e76ef   
200  732e4a29b2e318bce3c1e923276ddf51  594bec5e95e2ce005840c252   
584  c2dcc0b178176d7e6b185df9bd1cb4f2  594b33d6fee3760035a1a540   
297  0e9802516f8a79dd0d45211dd4ee74af  594aead5ef88150028311172   
..                                ...                       ...   
582  c2dcc0b178176d7e6b185df9bd1cb4f2  594b20ea3f82d10028f03c54   
96   62060850630c7afe54fd59151413d237  594c28953b593b00281250e1   
552  c2dcc0b178176d7e6b185df9bd1cb4f2  594b1abfa6992500289d6040   
530  c2dcc0b178176d7e6b185df9bd1cb4f2  594b31f09d87570035cccafd   
620  d9837e646a15a4fdd823531178b2d62e  594ab554c864e200288e76ef   
55   e2333b2dc03032f12c8526e45243f0c1  594c0b883b593b002812506e   
54   e2333b2dc03032f12c8526e45243f0c1  594c0b883b593b002812506e   
198  9286ce84f2401fbd6635f7d768c3cad1  594bec5e95e2ce005840c252   
208  d6eaa4eeb52b4d62ec78299454763870  594bec5c95e2ce005840c242   
633  85c7b78e76b5232cd38014ea4cdc8f56  594b20ea3f82d10028f03c54   
632  85c7b78e76b5232cd38014ea4cdc8f56  594b20ea3f82d10028f03c54   
342  a29fe6d26c1a49ff4a3c876eaee0a1af  594a9ce6c864e200288e766b   
259  9fa24ddce8fd9d1526d9d7451304fc74  5949a81c4d1319005556396c   
420  d6eaa4eeb52b4d62ec78299454763870  594b0303f3b9f600287eac2b   
732  98de7e62209a07eee6ec8dc984911042  594b06c06cdeef002881af83   
341  a29fe6d26c1a49ff4a3c876eaee0a1af  594a9ce6c864e200288e766b   
65   a09c8906073b4c0b75e3100b857b982a  594aead5ef88150028311172   
366  4bb6e496f0c92286dee7eed7038ddd9d  594b31f09d87570035cccafd   
196  9fa24ddce8fd9d1526d9d7451304fc74  594bec7ffd2cf400280aa953   
340  a29fe6d26c1a49ff4a3c876eaee0a1af  594a9ce6c864e200288e766b   
605  85c7b78e76b5232cd38014ea4cdc8f56  594b20eb3f82d10028f03c5c   
301  2df8ce7d317c7d89dfa95be7695d2de0  594b945bf8b866004036d9c3   
519  a09c8906073b4c0b75e3100b857b982a  594b20ec3f82d10028f03c64   
195  9fa24ddce8fd9d1526d9d7451304fc74  594bec7ffd2cf400280aa953   
114  98de7e62209a07eee6ec8dc984911042  594bec83fd2cf400280aa965   
713  d8e25a290ea51352bf9100a99c475f6d  59480d57e759070028da6467   
111  62060850630c7afe54fd59151413d237  594c1f5cd7e68f0028c9062c   
576  85c7b78e76b5232cd38014ea4cdc8f56  594b20ef3f82d10028f03c7c   
575  85c7b

**Longest Wait Time Per Analyst**

In [117]:
temp.groupby('Analyst')['Wait Time'].max().sort_values(ascending=False)

Analyst
f7f7591403c6c431053920223069550a   00:00:32
85c7b78e76b5232cd38014ea4cdc8f56   00:00:31
62060850630c7afe54fd59151413d237   00:00:29
98de7e62209a07eee6ec8dc984911042   00:00:27
d8e25a290ea51352bf9100a99c475f6d   00:00:27
9fa24ddce8fd9d1526d9d7451304fc74   00:00:26
a09c8906073b4c0b75e3100b857b982a   00:00:25
2df8ce7d317c7d89dfa95be7695d2de0   00:00:25
a29fe6d26c1a49ff4a3c876eaee0a1af   00:00:22
4bb6e496f0c92286dee7eed7038ddd9d   00:00:19
d6eaa4eeb52b4d62ec78299454763870   00:00:18
9286ce84f2401fbd6635f7d768c3cad1   00:00:16
e2333b2dc03032f12c8526e45243f0c1   00:00:16
d9837e646a15a4fdd823531178b2d62e   00:00:16
c2dcc0b178176d7e6b185df9bd1cb4f2   00:00:16
8258591d197a3936de0f7ee021ab9e0e   00:00:13
bcd8d747b5af397febc04228293f747f   00:00:13
710dbda77cade1eced9c735c08e6e38d   00:00:13
206de922289a1f9f5ee250fc71308628   00:00:13
de71db592b02c988670ca3d301a02568   00:00:11
9fcbc63ff4c8bea5cea4efad782c87cf   00:00:11
dddcdea13df8232803957a5d9c847aae   00:00:10
e8a2b0b2e2234a89dc52dd8a

In [132]:
data[data['Action']=='Accepted Job'].groupby('Request')['Action'].count()

Request
59480d57e759070028da6467     2
59485c262e71030033104e3c     5
59486e7af5874900429ce273     1
5949462dd9ae5200633f9640     3
59494c13d9ae5200633f9695     1
5949a81c4d1319005556396c    16
5949bdb4fe2f5d005b4f4910     1
5949bdbcfe2f5d005b4f4928     1
5949e376f03c130028baa724     2
594a4325499a400071d4cca4     1
594a536d220b450040dc0182     1
594a5e5566cb69006fdcf09d     7
594a8f5664251c002831f4c5     6
594a8fd422b3c50028e71976     8
594a9ce6c864e200288e766b    12
594aa36664251c002831f51e     1
594ab554c864e200288e76ef    12
594ada897a05b40035f08be4     2
594ae6944f06e20035d92325     3
594ae6994f06e20035d92335     1
594ae6fa4f06e20035d9233d     1
594ae6fb4f06e20035d92345     7
594aead5ef88150028311172     9
594aec513f82d10028f03b9a     1
594af29d3f82d10028f03bb2     8
594af3c6f3b9f600287eabe0     3
594b00e23f82d10028f03be2     4
594b0303f3b9f600287eac2b    10
594b06bf6cdeef002881af7b     1
594b06c06cdeef002881af83     7
                            ..
594babd780f13f00405cd449     1
